kvstore/memory.go

```diff
@@ -67,13 +67,14 @@ func (o *Memory) Set(key string, val string) error {
 	lk.Lock()
 	defer lk.Unlock()
 
-	_, ok := o.Data[key]
+	data, ok := o.Data[key]
 	if ok {
-		return fmt.Errorf("key %q already exist, delete it first", key)
+		// TODO(tho) -- check if the value is already stored, if so do not append
```

Comment:
Isn't the comment the opposite of what you are actually doing...?

Reply:
I think the comment is correct.  We are not checking if the value is already in the array at the moment, which is what the comment says.